packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -828,22 +828,51 @@ class RenderListWheelViewport
 
   /// Paints all children visible in the current viewport.
   void _paintVisibleChildren(PaintingContext context, Offset offset) {
+    if (overAndUnderCenterOpacity >= 1) {
+      RenderBox? childToPaint = firstChild;
+      while (childToPaint != null) {
+        final ListWheelParentData childParentData = childToPaint.parentData! as ListWheelParentData;
+        // center value is ignored if overAndUnderCenterOpacity >= 1.
+        _paintTransformedChild(childToPaint, context, offset, childParentData.offset, center: null);
+        childToPaint = childAfter(childToPaint);
+      }
+      return;
+    }
+
+    // In order to reduce the number of opacity layers, we first paint all
+    // partially opaque children, then finally paint the fully opaque children.
+    context.pushOpacity(offset, (overAndUnderCenterOpacity * 255).round(), (PaintingContext context, Offset offset) {
+      RenderBox? childToPaint = firstChild;
+      while (childToPaint != null) {
+        final ListWheelParentData childParentData = childToPaint.parentData! as ListWheelParentData;
+        _paintTransformedChild(childToPaint, context, offset, childParentData.offset, center: false);
+        childToPaint = childAfter(childToPaint);
+      }
+    });
+
     RenderBox? childToPaint = firstChild;
     while (childToPaint != null) {
       final ListWheelParentData childParentData = childToPaint.parentData! as ListWheelParentData;
-      _paintTransformedChild(childToPaint, context, offset, childParentData.offset);
+      _paintTransformedChild(childToPaint, context, offset, childParentData.offset, center: true);
       childToPaint = childAfter(childToPaint);
     }
   }
 
-  /// Takes in a child with a **scrollable layout offset** and paints it in the
-  /// **transformed cylindrical space viewport painting coordinates**.
+  // Takes in a child with a **scrollable layout offset** and paints it in the
+  // **transformed cylindrical space viewport painting coordinates**.
+  //
+  // If [center] is true, only children that overlap with the center section
+  // will be painted. If [center] is false, only children that do not overlap
+  // will be painted. Opacity is only applied to the partially overlapping
```

Comment:
Is this comment about Opacity here actually true? My understanding was that opacity is fully controlled from outside this method now? I.e. it is up to the caller to apply opacity where necessary.

Reply:
Updated